### ISSUE TITLE
Overriding flags set in wp-cli.yml does not work

### ISSUE THREAD

**Issue by JPry** [wp-cli/wp-cli#1696]
In the [Configuration Documentation](http://wp-cli.org/config/), there's the following sentence:

> Besides the global parameters described above, config files can also contain defaults for any subcommand.

I've tried setting some default flags for commands, and I'm finding that I cannot override those flags when I actually call the command on the command line.

Here's an example `wp-cli.yml`:

```yml
search-replace:
    precise: true
    dry-run: true
```

Then I run something like this:
```
wp search-replace foo bar --no-dry-run --no-precise
```

Instead of doing the search-replace as expected, it is a dry run with PHP instead of SQL replacement.

Am I doing this the wrong way, or is this a bug with default options?

**Comment by danielbachhuber** [comment]
I suspect a bug but haven't tracked it down. I've confirmed the unexpected behavior.

**Comment by szepeviktor** [comment]
https://github.com/wp-cli/wp-cli/pull/1363 would come handy.

**Comment by JPry** [comment]
Fixed by #1697

**Comment by danielbachhuber** [comment]
This actually a problem across many commands. We should come up with a more holistic fix.

**Comment by JPry** [comment]
@danielbachhuber It might be good to have some kind of general helper function for checking boolean flags, and then consistently use that instead of having to remember to do this:

```php
$foo = isset( $assoc_args['foo'] && $assoc_args['foo'] );
```

😉

**Comment by danielbachhuber** [comment]
> It might be good to have some kind of general helper function for checking boolean flags

I'm up for a pull request.

**Comment by danielbachhuber** [comment]
Fixed by #1732
